tests/system/aiplatform/test_e2e_forecasting.py

```diff
@@ -0,0 +1,124 @@
+            staging_bucket=shared_state["staging_bucket_name"],
+        )
+        try:
+            # Create and import to single managed dataset for both training
+            # jobs.
+            ds = aiplatform.TimeSeriesDataset.create(
+                display_name=self._make_display_name("dataset"),
+                bq_source=[_TRAINING_DATASET_BQ_PATH],
+                sync=False,
+                create_request_timeout=180.0,
+            )
+
+            time_column = "date"
+            time_series_identifier_column = "store_name"
+            target_column = "sale_dollars"
+            column_specs = {
+                time_column: "timestamp",
+                target_column: "numeric",
+                "city": "categorical",
+                "zip_code": "categorical",
+                "county": "categorical",
+            }
+
+            # Define both training jobs
+            # TODO(humichael): Add seq2seq job.
+            automl_job = aiplatform.AutoMLForecastingTrainingJob(
+                display_name=self._make_display_name("train-housing-automl"),
+                optimization_objective="minimize-rmse",
+                column_specs=column_specs,
+            )
+
+            # Kick off both training jobs, AutoML job will take approx one hour
+            # to run.
+            automl_model = automl_job.run(
+                dataset=ds,
+                target_column=target_column,
+                time_column=time_column,
+                time_series_identifier_column=time_series_identifier_column,
+                available_at_forecast_columns=[time_column],
+                unavailable_at_forecast_columns=[target_column],
+                time_series_attribute_columns=["city", "zip_code", "county"],
+                forecast_horizon=30,
+                context_window=30,
+                data_granularity_unit="day",
+                data_granularity_count=1,
+                budget_milli_node_hours=1000,
+                model_display_name=self._make_display_name("automl-liquor-model"),
+                sync=False,
+            )
+
+            automl_batch_prediction_job = automl_model.batch_predict(
+                job_display_name=self._make_display_name("automl-liquor-model"),
+                instances_format="bigquery",
+                machine_type="n1-standard-4",
+                bigquery_source=_PREDICTION_DATASET_BQ_PATH,
+                gcs_destination_prefix=(
+                    f'gs://{shared_state["staging_bucket_name"]}/bp_results/'
+                ),
+                sync=False,
+            )
+
+            automl_batch_prediction_job.wait()
+
+            assert (
+                automl_job.state
+                == pipeline_state.PipelineState.PIPELINE_STATE_SUCCEEDED
+            )
+            assert (
+                automl_batch_prediction_job.state
+                == job_state.JobState.JOB_STATE_SUCCEEDED
+            )
+        finally:
+            resources = [
```

Comment:
Prefer to avoid "locals" and just be explicit about the variables.
When we later integrate mypy, it might have trouble with the type inference.

Reply:
Just put the deletions on separate lines.

Also instantiate each variable to None before the `try` block.